Error on completion with spring-idea-plugin

When trying to trigger code completion for properties and yml files after enabling the extensions i get the following error. And not completions are shown 

```
Mar 17, 2019 9:59:32 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
java.util.concurrent.CompletionException: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1629)
	at java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1618)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at java.util.ArrayList.rangeCheck(ArrayList.java:653)
	at java.util.ArrayList.get(ArrayList.java:429)
	at com.github.gtache.lsp.client.DynamicRegistrationMethods.forName(DynamicRegistrationMethods.java:37)
	at com.github.gtache.lsp.client.languageserver.wrapper.LanguageServerWrapperImpl.$anonfun$registerCapability$2(LanguageServerWrapperImpl.scala:380)
	at scala.collection.Iterator.foreach(Iterator.scala:937)
	at scala.collection.Iterator.foreach$(Iterator.scala:937)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1425)
	at scala.collection.IterableLike.foreach(IterableLike.scala:70)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:69)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at com.github.gtache.lsp.client.languageserver.wrapper.LanguageServerWrapperImpl.$anonfun$registerCapability$1(LanguageServerWrapperImpl.scala:378)
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1626)
	... 5 more

Mar 17, 2019 9:59:32 PM org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint request
WARNING: Unsupported request method: sts/addClasspathListener
```

--- a/intellij-lsp/src/com/github/gtache/lsp/client/DynamicRegistrationMethods.java
+++ b/intellij-lsp/src/com/github/gtache/lsp/client/DynamicRegistrationMethods.java
@@ -1,7 +1,7 @@
 package com.github.gtache.lsp.client;
 
 import java.util.Arrays;
-import java.util.stream.Collectors;
+import java.util.Optional;
 
 /**
  * Enum for methods which may support DynamicRegistration
@@ -33,8 +33,8 @@ public enum DynamicRegistrationMethods {
         this.name = name;
     }
 
-    public static DynamicRegistrationMethods forName(final String name) {
-        return Arrays.stream(DynamicRegistrationMethods.values()).filter(n -> n.name.equals(name)).collect(Collectors.toList()).get(0);
+    public static Optional<DynamicRegistrationMethods> forName(final String name) {
+        return Arrays.stream(DynamicRegistrationMethods.values()).filter(n -> n.name.equals(name)).findFirst();
     }
 
     public String getName() {

--- a/intellij-lsp/src/com/github/gtache/lsp/client/languageserver/wrapper/LanguageServerWrapperImpl.scala
+++ b/intellij-lsp/src/com/github/gtache/lsp/client/languageserver/wrapper/LanguageServerWrapperImpl.scala
@@ -378,8 +378,10 @@ class LanguageServerWrapperImpl(val serverDefinition: LanguageServerDefinition, 
       params.getRegistrations.asScala.foreach(r => {
         val id = r.getId
         val method = DynamicRegistrationMethods.forName(r.getMethod)
-        val options = r.getRegisterOptions
-        registrations.put(id, method)
+        if(method.isPresent) {
+          val options = r.getRegisterOptions
+          registrations.put(id, method.get())
+        }
       })
     })
   }
@@ -390,12 +392,14 @@ class LanguageServerWrapperImpl(val serverDefinition: LanguageServerDefinition, 
       params.getUnregisterations.asScala.foreach(r => {
         val id = r.getId
         val method = DynamicRegistrationMethods.forName(r.getMethod)
-        if (registrations.contains(id)) {
-          registrations.remove(id)
-        } else {
-          val invert = registrations.map(mapping => (mapping._2, mapping._1))
-          if (invert.contains(method)) {
-            registrations.remove(invert(method))
+        if(method.isPresent) {
+          if (registrations.contains(id)) {
+            registrations.remove(id)
+          } else {
+            val invert = registrations.map(mapping => (mapping._2, mapping._1))
+            if (invert.contains(method.get())) {
+              registrations.remove(invert(method.get()))
+            }
           }
         }
       })

